camille/source/zephyre.py

```diff
@@ -0,0 +1,119 @@
+from io import StringIO
+from pytz import utc
+from subprocess import PIPE
+from subprocess import Popen
+import pandas as pd
+import requests
+
+
+urljoin = requests.compat.urljoin
+default_host='http://localhost:9010'
+
+
+def isoformat(date):
+    if date.tzinfo != utc:
+        raise ValueError('Dates must be UTC')
+    date = date.replace(tzinfo=None)
+    return date.isoformat(timespec="milliseconds") + 'Z'
+
+
+def parse_response(strio, tzinfo=utc):
+    df = pd.read_json(strio, orient='records')
+
+    if df is None or df.empty or 'time' not in df.columns:
+        df.drop(df.index, inplace=True)
+        df.insert(0, 'time', [])
+        df.insert(1, 'value', [])
+
+    df.value = pd.to_numeric(df.value)
+    df.time = pd.to_datetime(df.time)
+    df.set_index('time', inplace=True)
+    df.index = df.index.tz_localize(tzinfo)
+    df.sort_index(inplace=True)
+
+    return df.value
+
+
+class Zephyre:
+    """Zephyre
+
+    Callable object that can be used to fetch time series from zephyre service.
+
+    Notes
+    -----
+    `oauth2local serve` must be running in order to authenticate with the
+    zephyre API.
+
+    Examples
+    --------
+    Fetch time series named `tag`:
+
+    >>> start_date = utcdate(year=2019, month=1, day=1)
+    >>> end_date = utcdate(year=2019, month=1, day=1)
+    >>> z = Zephyre()
+    >>> ts = z('tag', start_date, end_date, snap='both')
+    """
+
+    def __init__(self, host=default_host):
+        self.host = host
+
+    def _get_token(self):
+        proc = Popen(['oauth2local', 'token'], stdin=PIPE, stdout=PIPE)
+        out, err = proc.communicate()
+        exit_code = proc.wait(timeout=0.005)
```

Comment:
Minor: Is this holding a thread or polling/async

Reply:
It _should_ be done after communicate, the timeout should be irrelevant